lib/charms/vault_k8s/v0/vault_managers.py

```diff
@@ -504,3 +550,304 @@ def generate_vault_unit_certificate(
 def existing_certificate_is_self_signed(ca_certificate: Certificate) -> bool:
     """Return whether the certificate is a self signed certificate generated by the Vault charm."""
     return ca_certificate.common_name == VAULT_CA_SUBJECT
+
+
+class VaultAutounsealNaming:
```

Comment:
We'll need such names for other features right? Ex for kv we may need a `charm-kv-<relation id>` policy (or something like this). Should we remove the Autounseal from the class name and allow for other features to use it as well?

Reply:
https://github.com/canonical/vault-k8s-operator/pull/534/commits/296e23fbc6af92cfe1d3aaabc0cf99bfb08cd2ab